pkg/registry/clusteroperator/clusterversion/strategy.go

```diff
@@ -0,0 +1,156 @@
+type clusterVersionStatusRESTStrategy struct {
+	clusterVersionRESTStrategy
+}
+
+var (
+	clusterVersionRESTStrategies = clusterVersionRESTStrategy{
+		// embeds to pull in existing code behavior from upstream
+
+		// this has an interesting NOTE on it. Not sure if it applies to us.
+		ObjectTyper: api.Scheme,
+		// use the generator from upstream k8s, or implement method
+		// `GenerateName(base string) string`
+		NameGenerator: names.SimpleNameGenerator,
+	}
+	_ rest.RESTCreateStrategy = clusterVersionRESTStrategies
+	_ rest.RESTUpdateStrategy = clusterVersionRESTStrategies
+	_ rest.RESTDeleteStrategy = clusterVersionRESTStrategies
+
+	clusterVersionStatusUpdateStrategy = clusterVersionStatusRESTStrategy{
+		clusterVersionRESTStrategies,
+	}
+	_ rest.RESTUpdateStrategy = clusterVersionStatusUpdateStrategy
+)
+
+// Canonicalize does not transform a cluster.
+func (clusterVersionRESTStrategy) Canonicalize(obj runtime.Object) {
+	_, ok := obj.(*clusteroperator.ClusterVersion)
+	if !ok {
+		glog.Fatal("received a non-cluster version object to create")
+	}
+}
+
+// NamespaceScoped returns true as cluster versions are scoped to a namespace.
+func (clusterVersionRESTStrategy) NamespaceScoped() bool {
+	return true
+}
+
+func (clusterVersionRESTStrategy) PrepareForCreate(ctx genericapirequest.Context, obj runtime.Object) {
+	cv, ok := obj.(*clusteroperator.ClusterVersion)
+	if !ok {
+		glog.Fatal("received a non-cluster version object to create")
+	}
+
+	// Creating a brand new object, thus it must have no
+	// status. We can't fail here if they passed a status in, so
+	// we just wipe it clean.
+	cv.Status = clusteroperator.ClusterVersionStatus{}
+}
+
+func (clusterVersionRESTStrategy) Validate(ctx genericapirequest.Context, obj runtime.Object) field.ErrorList {
+	return validation.ValidateClusterVersion(obj.(*clusteroperator.ClusterVersion))
+}
+
+func (clusterVersionRESTStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (clusterVersionRESTStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+func (clusterVersionRESTStrategy) PrepareForUpdate(ctx genericapirequest.Context, new, old runtime.Object) {
+	newCV, ok := new.(*clusteroperator.ClusterVersion)
+	if !ok {
+		glog.Fatal("received a non-cluster version object to update to")
+	}
+	oldCV, ok := old.(*clusteroperator.ClusterVersion)
+	if !ok {
+		glog.Fatal("received a non-cluster version object to update from")
+	}
+	newCV.Status = oldCV.Status
+}
+
+func (clusterVersionRESTStrategy) ValidateUpdate(ctx genericapirequest.Context, new, old runtime.Object) field.ErrorList {
+	_, ok := new.(*clusteroperator.ClusterVersion)
+	if !ok {
+		glog.Fatal("received a non-cluster version object to validate to")
+	}
+	_, ok = old.(*clusteroperator.ClusterVersion)
+	if !ok {
+		glog.Fatal("received a non-cluster version object to validate from")
+	}
+
+	return field.ErrorList{}
+	//return validation.ValidateClusterUpdate(newCluster, oldCluster)
```

Comment:
Why are we not doing any validation on updates?

Reply:
Will add this in a followup.

Reply:
Can we at least call `ValidateClusterVersion` here, even if we are not going to do any validation of changes until a follow-up?